pagopa/io-app
# chore: [IOBP-215] Unused locales scripts

## Short description
added a script that removes unused locales

## List of changes proposed in this pull request

this script:
- checks wether [`yq`](https://github.com/mikefarah/yq) is installed, if not it installs it 
- checks for unused locales through `grep` and a series of regExp shenanigans
- removes unused keys from all locales
- also small update to an util function, this breaks nothing since it's the addition of a possible param with base value

please note that it does not remove *ALL* of the unused keys, since in the app they are often referenced with a non-literal notation, so it tries to do a best guess by substituting the various yaml entries with a "catch all" char in grep (e.g `wallet.time.now` gets checked for `wallet.*.now` and `wallet.time.*`),
so if a single subentry is used in a key, then all are flagged as in use.

## How to test
run `yarn remove_unused_locales`, then `yarn generate:locales` and `yarn tsc:noemit`, and make sure nothing breaks. Then have a tour around the app to make sure no I18n is missing (although `yarn tsc:noemit` should already tell if any keys are missing from the codebase)
